R/sensitivity-tornado-plot.R

```diff
@@ -0,0 +1,304 @@
+#' single output path is specified, or a list of `patchwork` objects for
+#' multiple output paths.
+#'
+#' @examples
+#' \dontrun{
+#' simPath <- system.file("extdata", "Aciclovir.pkml", package = "ospsuite")
+#' simulation <- loadSimulation(simPath)
+#' outputPaths <- "Organism|PeripheralVenousBlood|Aciclovir|Plasma (Peripheral Venous Blood)"
+#' parameterPaths <- c(
+#'   "Aciclovir|Lipophilicity",
+#'   "Applications|IV 250mg 10min|Application_1|ProtocolSchemaItem|Dose",
+#'   "Neighborhoods|Kidney_pls_Kidney_ur|Aciclovir|Glomerular Filtration-GFR|GFR fraction"
+#' )
+#'
+#' results <- sensitivityCalculation(
+#'   simulation = simulation,
+#'   outputPaths = outputPaths,
+#'   parameterPaths = parameterPaths,
+#'   variationRange = c(seq(0.1, 1, by = 0.1), seq(2, 10, by = 1)),
+#' )
+#'
+#' # Print plots with default settings
+#' sensitivityTornadoPlot(results)
+#'
+#' # Print plots with specific parameter scaling factor
+#' sensitivityTornadoPlot(results, parameterFactor = 0.5)
+#'
+#' # Print plots with custom configuration settings
+#' myPlotConfiguration <- createEsqlabsPlotConfiguration()
+#' myPlotConfiguration$legendPosition <- "bottom"
+#' myPlotConfiguration$subtitle <- "Custom settings"
+#' sensitivityTornadoPlot(results, defaultPlotConfiguration = myPlotConfiguration)
+#' }
+#'
+#' @export
+sensitivityTornadoPlot <- function(sensitivityCalculation,
+                                   outputPaths = NULL,
+                                   parameterPaths = NULL,
+                                   pkParameters = NULL,
+                                   parameterFactor = 0.1,
+                                   defaultPlotConfiguration = NULL) {
+  # input validation -------------------------
+
+  # fail early if the object is of wrong type
+  validateIsOfType(sensitivityCalculation, "SensitivityCalculation")
+  if (is.null(defaultPlotConfiguration)) {
+    defaultPlotConfiguration <- createEsqlabsPlotConfiguration()
+  } else {
+    validateIsOfType(defaultPlotConfiguration, "DefaultPlotConfiguration")
+  }
+  ospsuite.utils::validateIsOption(
+    list(parameterFactor = parameterFactor),
+    .getPlotConfigurationOptions("parameterFactor")
+  )
+
+  # validate vector arguments of character type
+  .validateCharVectors(outputPaths)
+  .validateCharVectors(parameterPaths)
+  .validateCharVectors(pkParameters)
+
+  data <- sensitivityCalculation$pkData
+
+  # validate data contains required parameterFactor results
+  parameterFactors <- c(parameterFactor, 1 / parameterFactor)
+  if (!all(parameterFactors %in% data$ParameterFactor)) {
+    stop(messages$noParameterFactor(data, parameterFactor))
+  }
+
+  # default tornado plot configuration setup ----
+
+  tornadoPlotConfiguration <- list(
+    legendPosition = "right",
+    legendTitle = "Parameter Factor",
+    subtitle = NULL,
+    title = NULL,
+    titleSize = 14,
+    xLabel = "Parameter",
+    yLabel = "Input parameter value [% of reference]"
```

Comment:
I think this will be confusing, because of the cord_flip, the yLabel ends up in the final x-axis. Can we also flip the label such that what the user give as xAxis is the final xAxis. I know this is the default but it would also apply to the given plot configuration.

Reply:
Indeed, this is confusing. Swapped labels as suggested